underscore.js

```diff
@@ -779,13 +779,21 @@
 
   // Memoize an expensive function by storing its results.
   _.memoize = function(func, hasher) {
+    function getCache() {
+      if (Object.create) {
+        return Object.create(null);
+      }
+      return {};
+    }
+
     var memoize = function(key) {
       var cache = memoize.cache;
       var address = '' + (hasher ? hasher.apply(this, arguments) : key);
-      if (!_.has(cache, address)) cache[address] = func.apply(this, arguments);
+      if (!(_.has(cache, address))) cache[address] = func.apply(this, arguments);
       return cache[address];
```

Comment:
Hey @caiogondim! I created a [screencast](https://www.dropbox.com/s/ykq7cnormkkbc9f/MapCache.mp4?dl=0) to dig into `MapCache` a bit more. You can follow along at [4.6.0-npm-packages/lodash._mapcache/index.js](https://github.com/lodash/lodash/blob/4.6.0-npm-packages/lodash._mapcache/index.js).